v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compilation-job.ts

```diff
@@ -20,6 +21,10 @@ export class CompilationJobImplementation implements CompilationJob {
 
   #buildId: string | undefined;
   #solcInput: CompilerInput | undefined;
+  #solcInputWithoutSources: Omit<CompilerInput, "sources"> | undefined;
+  #resolvedFiles: ResolvedFile[] | undefined;
```

Comment:
By adding these two new properties on the compilation job, we ensure they are only ever computed once per compilation job.